extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -909,27 +926,35 @@ private static String getVideoInfoUrl(final String id, final String sts) {
                 "&sts=" + sts + "&ps=default&gl=US&hl=en";
     }
 
-    private Map<String, ItagItem> getItags(String streamingDataKey, ItagItem.ItagType itagTypeWanted) throws ParsingException {
-        Map<String, ItagItem> urlAndItags = new LinkedHashMap<>();
-        JsonObject streamingData = playerResponse.getObject("streamingData");
+    private Map<String, ItagItem> getItags(final String streamingDataKey, final ItagItem.ItagType itagTypeWanted) throws ParsingException {
+        final Map<String, ItagItem> urlAndItags = new LinkedHashMap<>();
+        final JsonObject streamingData = playerResponse.getObject("streamingData");
         if (!streamingData.has(streamingDataKey)) {
             return urlAndItags;
         }
 
-        JsonArray formats = streamingData.getArray(streamingDataKey);
-        for (int i = 0; i != formats.size(); ++i) {
-            JsonObject formatData = formats.getObject(i);
-            int itag = formatData.getInt("itag");
+        final JsonArray formats = streamingData.getArray(streamingDataKey);
+        for (final Object format : formats) {
+            final JsonObject formatData = (JsonObject) format;
+            final int itag = formatData.getInt("itag");
 
             if (ItagItem.isSupported(itag)) {
                 try {
-                    ItagItem itagItem = ItagItem.getItag(itag);
+                    final ItagItem itagItem = ItagItem.getItag(itag);
                     if (itagItem.itagType == itagTypeWanted) {
+                        // Ignore streams that are delivered using YouTube's OTF format,
+                        // as those only work with DASH and not with progressive HTTP.
+                        // Those should be extracted when parsing the DASH MPD
+                        if (formatData.getString("type", EMPTY_STRING)
+                                .equalsIgnoreCase("FORMAT_STREAM_TYPE_OTF")) {
+                            continue;
+                        }
```

Comment:
Note: from the research I did a while ago it it theoretically possible to generate the manifest content based on this URL, without having to fetch the manifest with another time-consuming request

Reply:
I know, plus we don't need the whole `ItagItem` class, but I'll leave that to a future PR.